### ISSUE TITLE
Inline code is not syntax highlighted

### ISSUE THREAD

**Issue by jessetan** [sphinx-doc/sphinx#5157]
### Problem

It is not possible to apply syntax highlighting to inline literals or code.

The docutils documentation provides a way to do this using the `role` directive to define a new role, based on the `code` role, but the resulting HTML contains classes like `.keyword` instead of classes like `.highlight .k` which are used in Pygments CSS.

#### Expected results

Sphinx to provide a `code` role that applies proper classes for Pygments to style.

There is [an extension to add this](https://github.com/sphinx-contrib/inlinesyntaxhighlight), but it is from 2013 (so maybe out of date with recent Sphinx), and it does not contain code to generate highlighting in LaTeX.

### Reproducible project / your project

```rst
.. role:: js(code)
    :language: js

This is inline code :js:`function a() { return 1; }`
```

### Environment info

- Sphinx version: 1.7.5

**Comment by amerlyq** [comment]
Does `docutils.conf` workaround from here works ?
https://stackoverflow.com/questions/21591107/sphinx-inline-code-highlight

**Comment by jessetan** [comment]
Unfortunately it doesn't. That correctly applies the short class names in HTML and syntax highlighting works if youalso add the `highlight` class to the `<code>` element that contains the code.

It does not work for LaTeX output.
This:
```rst
.. role:: js(code)
    :language: js

:js:`window.addEventListener(function() { return false; });`

.. code-block:: js

    window.addEventListener(function() { return false; });
```
is converted to:

```tex
\sphinxcode{\sphinxupquote{\DUrole{nb}{window}\DUrole{p}{.}\DUrole{nx}{addEventListener}\DUrole{p}{(}\DUrole{kd}{function}\DUrole{p}{()} \DUrole{p}{\{} \DUrole{k}{return} \DUrole{kc}{false}\DUrole{p}{;} \DUrole{p}{\});}}}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n+nb}{window}\PYG{p}{.}\PYG{n+nx}{addEventListener}\PYG{p}{(}\PYG{k+kd}{function}\PYG{p}{(}\PYG{p}{)} \PYG{p}{\PYGZob{}} \PYG{k}{return} \PYG{k+kc}{false}\PYG{p}{;} \PYG{p}{\PYGZcb{}}\PYG{p}{)}\PYG{p}{;}
\end{sphinxVerbatim}
```

Only the code block (not the inline code) has highlighting

**Comment by tk0miya** [comment]
Yes, current version of Sphinx does not support inline highlighting on LaTeX output. So we need to improve LaTeX writer (or pygments bridge?) to support it.

cc: @jfbu

**Comment by jessetan** [comment]
I've created a PR with a possible implementation. Please review or rewrite as necessary.

**Comment by gmilde** [comment]
Could adding `'syntax_highlight': 'short'` to the `default_settings` in `sphinx/environment/__init__.py` solve this problem for HTML? (cf.  #9701)

This would still need the "highlight" class to work with the current CSS styles. 
Example from Stackexchange:
~~~
.. role:: py(code)
   :language: py
   :class: highlight
~~~
If this is too cumbersome, a change in the CSS selectors or a patch that adds "highlight" to Code nodes ("literal" nodes with "code" class in the Docutils document tree) may help.

For inline syntax highlight in LaTeX, definitions for "\DUrolenx", "\DUrolekd", ... are required.
See https://docutils.sourceforge.io/docs/user/latex.html#custom-interpreted-text-roles and the example https://docutils.sourceforge.io/sandbox/stylesheets/pygments-default.sty.

**Comment by tk0miya** [comment]
Fixed by #10251. It will be released as v5.0.
